<?xml version="1.0" encoding="iso-8859-15"?>

<launch>
  <arg name="launch-prefix" default=""/>
  <arg name="n_events_per_window" default="30000"/>
  <arg name="start_time" default="9.5"/>

  <arg name="bag_folder_path" default="/mnt/data/mvsec_data/indoor_flying"/>
  <arg name="sequence" default="indoor_flying1"/>
  <arg name="calib_sequence" default="indoor_flying"/>
  <arg name="data_bag_path" default="$(arg bag_folder_path)/$(arg sequence)_data.bag"/>
  <arg name="gt_bag_path" default="$(arg bag_folder_path)/$(arg sequence)_gt.bag"/>
  <arg name="calibration_file" default="$(arg bag_folder_path)/camchain-imucam-$(arg calib_sequence).yaml"/>

  <node pkg="event_cpp_python_node" type="event_cpp_python_node" name="event_cpp_node" output="screen" required="true" launch-prefix="$(arg launch-prefix)">
    <rosparam command="load" file="$(arg calibration_file)"/>
    <param name="data_bag_path" type="str" value="$(arg data_bag_path)"/>
    <param name="gt_bag_path" type="str" value="$(arg gt_bag_path)"/>
    <param name="n_events_per_window" type="int" value="$(arg n_events_per_window)"/>
    <param name="start_time" type="double" value="$(arg start_time)"/>
  </node>

  <node pkg="event_cpp_python_node" type="event_cpp_python_pynode.py" name="event_python_node" required="true" output="screen">
    <param name="n_events_per_window" type="int" value="$(arg n_events_per_window)"/>
    <param name="bag_folder_path" type="str" value="$(arg bag_folder_path)"/>
    <param name="sequence" type="str" value="$(arg calib_sequence)"/>
    </node>
</launch>
